jupyterlab_git/git.py

```diff
@@ -1092,16 +1116,35 @@ def remote_add(self, top_repo_path, url, name=DEFAULT_REMOTE_NAME):
             Remote name; default "origin"
         """
         cmd = ["git", "remote", "add", name, url]
-        p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=top_repo_path)
-        _, my_error = p.communicate()
-        if p.returncode == 0:
-            return {
-                "code": p.returncode,
+        code, _, error = await execute(cmd, cwd=top_repo_path, logger=self.logger)
+        response = {
+                "code": code,
                 "command": " ".join(cmd)
             }
-        else:
-            return {
-                "code": p.returncode,
-                "command": " ".join(cmd),
-                "message": my_error.decode("utf-8").strip()
+        if code != 0:
+            response["message"] = error
+
+        return response
+
+    async def remote_show(self, path):
+        """Handle call to `git remote show` command.
+
+        Args:
+            path (str): Git repository path
+        
+        Returns:
+            List[str]: Known remotes
+        """
+        command = ["git", "remote", "show"]
```

Comment:
Does it make sense to instead use `git remote -v show` here? As some future proofing of this function in case it is ever used to determine URLs or do something with determining fetch vs push urls?
For me `git remote show`:
```
origin
upstream
```

vs `git remote -v show`:
```
origin	git@github.com:ianhi/jupyterlab-git.git (fetch)
origin	git@github.com:ianhi/jupyterlab-git.git (push)
upstream	git@github.com:jupyterlab/jupyterlab-git.git (fetch)
upstream	git@github.com:jupyterlab/jupyterlab-git.git (push)
```

Reply:
For now I'll stick to current behavior as there is no feature requiring more information.